### PR TITLE
refactor(hubio): push, login, docker auth

### PR DESCRIPTION
**Pull request by deepankarm**

### CODE REVIEW

File: jina/docker/hubapi/local.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    namespace = 'jina.hub'
+    try:
+        path = os.path.dirname(pkgutil.get_loader(namespace).path)
+    except AttributeError:
+        default_logger.warning('local Hub is not initialized, '
+                               'try "git submodule update --init" if you are in dev mode')
+        return {}
+
+    def add_hub():
+        m_yml = f'{info.module_finder.path}/manifest.yml'
```
Comment (JoanFM) [inline jina/docker/hubapi/local.py:26]: os.path.join I think is better

File: jina/docker/hubapi/local.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+def _load_local_hub_manifest():
+    namespace = 'jina.hub'
+    try:
+        path = os.path.dirname(pkgutil.get_loader(namespace).path)
+    except AttributeError:
+        default_logger.warning('local Hub is not initialized, '
+                               'try "git submodule update --init" if you are in dev mode')
+        return {}
+
+    def add_hub():
```
[the remark this answers is not in the document]
Comment (JoanFM) [reply jina/docker/hubapi/local.py:25]: I actually I think is more elegant to have `get_hub` and have it return a key, value tuple to put in ur `hub_images` and so u do not pass `hub_images` as argument.

File: jina/docker/hubapi/local.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        with open(credentials_file(), 'r') as cf:
+            cred_yml = JAML.load(cf)
+            access_token = cred_yml['access_token']
+            return access_token
+    except KeyError:
+        logger.error(f'Invalid access file. '
+                     f'please re-login using command: {colored("jina hub login", attrs=["bold"])}')
+        raise HubLoginRequired
+
+def _make_hub_table_with_local(manifests, local_manifests):
```
Comment (JoanFM) [inline jina/docker/hubapi/local.py:83]: I prefer the wording `_get`

File: jina/docker/hubapi/local.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                else:
+                    color = 'yellow'
+            info_table.append(f'{colored(image_name, color="yellow", attrs="bold"):<50s}'
+                              f'{colored(ver, color="green"):<20s}'
+                              f'{colored(local_ver, color=color):<20s}'
+                              f'{desc:<30s}')
+    return info_table
+
+
+def _make_hub_table(manifests):
```
Comment (JoanFM) [inline jina/docker/hubapi/local.py:112]: same here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/docker/hubapi/local.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (deepankarm) [reply jina/docker/hubapi/local.py:25]: This is restructuring of the existing code. I would do this in the next PR, while fixing `list`

File: jina/docker/hubapi/local.py (hunk 1 above)
Comment (deepankarm) [reply jina/docker/hubapi/local.py:26]: Same.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**